graphql_relay/connection/arrayconnection.py

```diff
@@ -104,6 +104,121 @@ def connection_from_list_slice(list_slice, args=None, connection_type=None,
     )
 
 
+def connection_from_list_slice_lazy(list_slice, args=None, connection_type=None,
+                                    edge_type=None, pageinfo_type=None,
+                                    slice_start=0, list_length=0, list_slice_length=None):
+    '''
+    Given an iterator it consumes the needed amount based on the pagination
+    params, and also tries to figure out if there are more results to be
+    consumed. We do so by trying to fetch one more element than the specified
+    amount, if we were able to fetch n+1 it means there are more to be consumed.
+
+    This spares the caller passing the total count of results, which
+    usually means making an extra query just to find out that number.
+    '''
+    connection_type = connection_type or Connection
+    edge_type = edge_type or Edge
+    pageinfo_type = pageinfo_type or PageInfo
+
+    args = args or {}
+
+    before = args.get('before')
+    after = args.get('after')
+    first = args.get('first')
+    last = args.get('last')
+
+    first_is_negative = type(first) == int and first < 0
+    last_is_negative = type(last) == int and last < 0
+
+    start_offset = slice_start
+    end_offset = None
+
+    # Validations
+    if first_is_negative or last_is_negative:
+        raise ValueError("first and last can't be negative values")
+
+    if first and last:
+        raise ValueError(
+            "Including a value for both first and last is strongly discouraged,"
+            " as it is likely to lead to confusing queries and results"
+        )
+
+    if before and last is None:
+        raise ValueError(
+            "before without last is not supported"
+        )
+
+    if after and first is None:
+        raise ValueError(
+            "after without first is not supported"
+        )
+
+    if before and after:
+        raise ValueError(
+            "Mixing before and after is not supported"
+        )
+
+    if type(first) == int:
+        after_offset = get_offset_with_default(after, -1)
+        start_offset = after_offset + 1
+        end_offset = start_offset + first
+
+    elif type(last) == int:
+        if before:
+            before_offset = get_offset_with_default(before)
+        else:
+            try:
+                before_offset = int(list_slice.count())
```

Comment:
Isn't this defeating the purpose of this PR, since providing `last` would trigger a `count`?

There is likely no workaround, but I believe this should at least be made explicit that `first` and `last` won't exhibit symmetrical behaviors.